mithril-client/src/file_downloader/retry.rs

```diff
@@ -0,0 +1,268 @@
+use std::{path::Path, sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use mithril_common::{entities::CompressionAlgorithm, StdResult};
+
+use super::{DownloadEvent, FileDownloader, FileDownloaderUri};
+
+/// Policy for retrying file downloads.
+#[derive(Debug, PartialEq, Clone)]
+pub struct FileDownloadRetryPolicy {
+    /// Number of attempts to download a file.
+    pub attempts: usize,
+    /// Delay between two attempts.
+    pub delay_between_attempts: Duration,
+}
+
+impl FileDownloadRetryPolicy {
+    /// Create a policy that never retries.
+    pub fn never() -> Self {
+        Self {
+            attempts: 1,
+            delay_between_attempts: Duration::from_secs(0),
+        }
+    }
+}
+
+impl Default for FileDownloadRetryPolicy {
+    /// Create a default retry policy.
+    fn default() -> Self {
+        Self {
+            attempts: 3,
+            delay_between_attempts: Duration::from_secs(5),
+        }
+    }
+}
+
+/// RetryDownloader is a wrapper around FileDownloader that retries downloading a file if it fails.
+pub struct RetryDownloader {
+    /// File downloader to use.
+    file_downloader: Arc<dyn FileDownloader>,
+    /// Number of attempts to download a file.
+    pub retry_policy: FileDownloadRetryPolicy,
```

Comment:
I'm not sure that this field needs to be public.
```suggestion
    retry_policy: FileDownloadRetryPolicy,
```

Reply:
You're right, i made it private instead :+1: